packages/components/table/src/store/index.ts

```diff
@@ -119,7 +118,7 @@ function useStore<T>() {
       updateColumnOrder: () => void
     ) {
       const array = unref(states._columns) || []
-      if (parent) {
+      if (parent?.children) {
         parent.children.splice(
```

Comment:
This change may break the original logic, so I think it is best to keep this PR as is.

```suggestion
      if (parent) {
        parent.children?.splice(
```